exchangelib/properties.py

```diff
@@ -117,33 +117,40 @@ def from_xml(cls, elem, account):
         elem.clear()
         return cls(**kwargs)
 
-    def to_xml(self, version):
+    def to_xml(self, version, exclude_read_only=True):
         self.clean(version=version)
         # WARNING: The order of addition of XML elements is VERY important. Exchange expects XML elements in a
         # specific, non-documented order and will fail with meaningless errors if the order is wrong.
 
         # Call create_element() without args, to not fill up the cache with unique attribute values.
         elem = create_element(self.request_tag())
 
-        # Exchangelib previously skipped read_only fields while looping through both the
-        # attribute_fields and supported_fields below, but Nylas likes to keep them to surface fields like
-        # `is_cancelled`, `organizer`, and `item_id` in event raw_data.
+        # Exchangelib skips read_only fields while looping through both the attribute_fields and supported_fields
+        # by default, but Nylas likes to keep them to surface fields like `is_cancelled`, `organizer`, and `item_id`
+        # in event raw_data. That's the purpose of the `exclude_read_only` parameter.
 
         # Add attributes
         for f in self.attribute_fields():
+            if f.is_read_only and exclude_read_only is True:
+                continue
             value = getattr(self, f.name)
             if value is None or (f.is_list and not value):
                 continue
             elem.set(f.field_uri, value_to_xml_text(getattr(self, f.name)))
 
         # Add elements and values
         for f in self.supported_fields(version=version):
+            if f.is_read_only and exclude_read_only is True:
+                continue
             value = getattr(self, f.name)
             if value is None or (f.is_list and not value):
                 continue
             set_xml_value(elem, f.to_xml(value, version=version), version)
         return elem
 
+    def to_xml_read_only(self, version):
```

Comment:
Also minor, but do we really need this extra method versus just calling `to_xml` with the kwarg when we need to? (`to_xml_read_only` also sounds more like you're getting a read-only version of the xml rather than including read-only values)

Reply:
great point! i'm going to remove this method :)